src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1525,6 +1525,14 @@ private static bool WriteScriptBlockToLog(ScriptBlock scriptBlock, int segment,
                 // So, replace null characters with the Unicode `SYMBOL FOR NULL`
                 // We don't just remove the characters to preserve the fact that a null character was there.
                 textToLog = textToLog.Replace('\u0000', '\u2400');
+#if UNIX
+                if (Platform.IsLinux)
+                {
```

Comment:
This check is not needed.

Reply:
Why?  This is only an issues for SysLog, not OsLog on macOS

Reply:
Got it. Closed

Reply:
In the case it would be good to have a protection comment in the code.

Reply:
I can repeat the comment if you want

Reply:
It is explained on line 1584